lsp-metals.el

```diff
@@ -192,15 +193,46 @@ Should be ignored if there is no open doctor window."
     (setq lsp-metals--current-buffer (current-buffer))
     (lsp-notify "metals/didFocusTextDocument" (lsp--buffer-uri))))
 
+(declare-function dap-debug "ext:dap-mode" (conf))
+(declare-function dap-register-debug-provider "ext:dap-mode" (name conf))
+
+(defun lsp-metals--debug-start (no-debug params)
+  (dap-register-debug-provider "scala" #'identity)
+  (-let [session-params (lsp-send-execute-command
+                         "debug-adapter-start"
+                         (gethash "arguments" params))]
+    (dap-debug
+     (list :debugServer (-> (gethash "uri" session-params)
+                            (split-string ":")
+                            cl-third
+                            string-to-number)
+           :type "scala"
+           :name (gethash "name" session-params)
+           :host "localhost"
+           :request "launch"
+           :noDebug no-debug))))
```

Comment:
Just asking but can we do something similar to `lsp-rust` for `:action-handlers`?
``` elisp
:action-handlers (ht ("rls.run" 'lsp-rust--rls-run))
```
Then we can define a `cl-defmethod` for `metals-debug-session-start` and `metals-run-session-start` inside `dap-metals`.
That way the `lsp-mode` doesn't need to be aware of `dap-mode` and we can keep clean dependency between `lsp-mode` and `dap-mode`.

Reply:
We can achieve that in multiple ways, e. g. editing the client itself - ATM we are having a discussion to move lsp-metals in a separate package to avoid that kind of external deps so I went for the simplest solution .